src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -121,44 +129,91 @@ export class DomEventPlugin implements Plugin {
         return '';
     }
 
-    private addEventHandler(domEvent: TargetDomEvent): void {
+    private addEventHandler(
+        domEvent: TargetDomEvent,
+        targetElement?: HTMLElement
+    ): void {
         const eventType = domEvent.event;
         const eventListener = this.getEventListener(domEvent.cssLocator);
-        this.eventListenerMap.set(domEvent, eventListener);
+        const elementsList = [];
 
         // first add event listener to all elements identified by the CSS locator
         if (domEvent.cssLocator) {
+            let element: HTMLElement;
             const elementList = document.querySelectorAll(domEvent.cssLocator);
-            elementList.forEach((element: HTMLElement) => {
-                element.addEventListener(eventType, eventListener);
+            elementList.forEach((identifiedElement: HTMLElement) => {
+                if (targetElement) {
+                    if (targetElement === identifiedElement) {
+                        element = targetElement;
+                    }
+                } else {
+                    element = identifiedElement;
+                }
+
+                if (element) {
```

Comment:
question: Is `element` guaranteed to by truthy at this point?

Reply:
Clarified over call about why it doesn't have to be truthy. Also mentioned that this is insignificant with refactoring.